### PR TITLE
Add artifact builder script

### PR DESCRIPTION
**Pull request by jtattermusch**
This is work in progress
-- doesn't support docker yet
-- some builds are failing currently

Output can be seen here: https://grpc-testing.appspot.com/job/gRPC_build_artifacts/

### CODE REVIEW

File: tools/run_tests/build_artifacts.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env python
+# Copyright 2015-2016, Google Inc.
```
Comment (ctiller) [inline tools/run_tests/build_artifacts.py]: Just 2016 right?

**Comment by ctiller** [comment]
LGTM so far

File: tools/run_tests/build_artifacts.py (hunk 1 above)
Comment (jtattermusch) [inline tools/run_tests/build_artifacts.py]: Done.

**Comment by jtattermusch** [comment]
This now supports building on linux (using docker), mac and windows. 

On windows, collection of the artifacts doesn't work yet, but I can address that in a followup PR.

Ready for review.

**Pull request merged**